tools/bus_space/bus.c

```diff
@@ -156,9 +156,12 @@ bs_read(int rid, off_t ofs, void *buf, ssize_t bufsz)
 		case 4:
 			*((uint32_t *)buf) = *((volatile uint32_t *)ptr);
 			break;
+		case 8:
+			*((uint64_t *)buf)[] = *((volatile uint64_t *)ptr)[];
```

Comment:
remove the '[]'. It won't compile with them there.

Reply:
On amd64 it compiles. Instead of dimension N then you have dimension 1 when removing the []. Can you send me the testbuild logs, then I'll check it.

Reply:
it surprises me that it builds.... It is semantically incorrect` even if it does. buf points to the 8 byte thing you want to interact with, not to an array of pointers to such things.

Reply:
```
===> C (all)
cc  -O2 -pipe -fno-common -I/usr/src/tools/bus_space/C/..   -g -gz=zlib -MD  -MF.depend.bus.o -MTbus.o -std=gnu99 -Wno-format-zero-length -fstack-protector-strong -Wsystem-headers -Werror -Wall -Wno-format-y2k -W -Wno-unused-parameter -Wstrict-prototypes -Wmissing-prototypes -Wpointer-arith -Wreturn-type -Wcast-qual -Wwrite-strings -Wswitch -Wshadow -Wunused-parameter -Wcast-align -Wchar-subscripts -Wnested-externs -Wold-style-definition -Wno-pointer-sign -Wdate-time -Wmissing-variable-declarations -Wthread-safety -Wno-empty-body -Wno-string-plus-int -Wno-unused-const-variable -Wno-error=unused-but-set-variable  -Qunused-arguments    -c /usr/src/tools/bus_space/C/../bus.c -o bus.o
/usr/src/tools/bus_space/C/../bus.c:160:23: error: expected expression
                        *((uint64_t *)buf)[] = *((volatile uint64_t *)ptr)[];
                                           ^
/usr/src/tools/bus_space/C/../bus.c:160:55: error: expected expression
                        *((uint64_t *)buf)[] = *((volatile uint64_t *)ptr)[];
                                                                           ^
/usr/src/tools/bus_space/C/../bus.c:261:32: error: expected expression
                        *((volatile uint64_t *)ptr)[] = *((uint64_t *)buf)[];
                                                    ^
/usr/src/tools/bus_space/C/../bus.c:261:55: error: expected expression
                        *((volatile uint64_t *)ptr)[] = *((uint64_t *)buf)[];
                                                                           ^
4 errors generated.
*** Error code 1

Stop.
make[1]: stopped in /usr/src/tools/bus_space/C
*** Error code 1

Stop.
make: stopped in /usr/src/tools/bus_space
```

Reply:
Yea, that's the error that I thought should happen.... The [] need to be removed for this pull request to move forward.